plugins/modules/external_user_mappings.py

```diff
@@ -0,0 +1,337 @@
+# Copyright 2024 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerModule,
+)
+
+from cm_client import (
+    ExternalUserMappingsResourceApi,
+    ApiExternalUserMapping,
+    ApiAuthRoleRef,
+)
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: external_user_mappings
+short_description: Create, update, or delete external user mappings
+description:
+  - Configure details of a specific external user mapping.
+  - Create a new external user mapping.
+  - Update an existing external user mapping.
+  - Delete a external user mapping.
+  - The module supports C(check_mode).
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+options:
+  name:
```

Comment:
Need to have "required one of" logic for `name` or `uuid`, so `required_one_of` plus `mutually_exclusive` in the argument spec.